src/modules.ts

```diff
@@ -36,19 +37,19 @@ export async function generateInjectManifest(options: ResolvedVitePWAOptions, vi
   // self.__WB_MANIFEST is default injection point
   precacheAndRoute(self.__WB_MANIFEST)
   */
+  const vitePlugins = options.vitePlugins
+  const includedPluginNames: string[] = []
+  if (typeof vitePlugins === 'function')
+    includedPluginNames.push(...vitePlugins(viteOptions.plugins.map(p => p.name)))
+  else
+    includedPluginNames.push(...vitePlugins)
+
+  if (includedPluginNames.length === 0)
+    includedPluginNames.push(...defaultInjectManifestVitePlugins)
+
+  const plugins = viteOptions.plugins.filter(p => includedPluginNames.includes(p.name)) as Plugin[]
```

Comment:
BTW, why this type assertion? You're not actually importing the rollup Plugin type. You're asserting here that these are the deprecated browser plugin type from lib.dom.d.ts, which is weird and wrong. TS already knows that they are of the correct type.